.github/workflows/main.yml

```diff
@@ -45,6 +45,10 @@ jobs:
             shell: msys2 {0}
             env_cc: clang
             env_ar: llvm-ar
+          - os: ubuntu-20.04
+            java_version: 11
+            scala_version: 2.12.14
+            sonarscan: true
```

Comment:
I wonder if we could do away with the extra include and simply add the sonarscan to the existing ubuntu-20.04 include.  For that to work, we would need to define the sonarscan using an expression instead of a literal:

            sonarscan: ${{ matrix.java_version == 11 }}

If that expression works, we could make the if statements even more concise by moving their remaining expression clauses into sonarscan's expression too:

            sonarscan: ${{ matrix.java_version == 11 && (github.event_name == 'push' || github.event.pull_request.head.repo.full_name == 'apache/daffodil') }}
            # or even shorter...
            sonarscan: ${{ matrix.java_version == 11 && secrets.SONAR_TOKEN != '' }}

I have no idea whether these expressions would work, but it's worth a try.

Reply:
I'll give it a shot and see what happens. Note the we should also check the scala_version in case we ever start cross compiling and add another scala version to the matrix.